website/content/v1.0/talos-guides/install/cloud-platforms/vultr.md

```diff
@@ -5,4 +5,147 @@ aliases:
+## Creating a Cluster using the Vultr CLI
+
+This guide will demonstrate how to create a highly-available Kubernetes cluster with one worker using the Vultr cloud provider.
+
+[Vultr](https://www.vultr.com/) have a very well documented REST API, and an open-source [CLI](https://github.com/vultr/vultr-cli) tool to interact with the API which will be used in this guide.
+Make sure to follow installation and authentication instructions for the `vultr-cli` tool.
+
+### Upload image
+
+First step is to make the Talos ISO available to Vultr by uploading the latest release of the ISO to the Vultr ISO server.
+
+```bash
+vultr-cli iso create --url https://github.com/siderolabs/talos/releases/download/{{< release >}}/talos-amd64.iso
+```
+
+Make a note of the `ID` in the output, it will be needed later when creating the instances.
+
+### Create a Load Balancer
+
+A load balancer is needed to serve as the Kubernetes endpoint for the cluster.
+
+```bash
+vultr-cli load-balancer create \
+   --region $REGION \
+   --label "Talos Kubernetes Endpoint" \
+   --port 6443 \
+   --protocol tcp \
+   --check-interval 10 \
+   --response-timeout 5 \
+   --healthy-threshold 5 \
+   --unhealthy-threshold 3 \
+   --forwarding-rules frontend_protocol:tcp,frontend_port:443,backend_protocol:tcp,backend_port:6443
+```
+
+Make a note of the `ID` of the load balancer from the output of the above command, it will be needed after the control plane instances are created.
+
+```bash
+vultr-cli load-balancer get $LOAD_BALANCER_ID | grep ^IP
+```
+
+Make a note of the `IP` address, it will be needed later when generating the configuration.
+
+### Create the Machine Configuration
+
+#### Generate Base Configuration
+
+Using the IP address (or DNS name if one was created) of the load balancer created above, generate the machine configuration files for the new cluster.
+
+```bash
+talosctl gen config talos-kubernetes-vultr https://$LOAD_BALANCER_ADDRESS
+```
+
+Once generated, the machine configuration can be modified as necessary for the new cluster, for instance updating disk installation, or adding SANs for the certificates.
+
+#### Validate the Configuration Files
+
+```bash
+talosctl validate --config controlplane.yaml --mode cloud
+talosctl validate --config worker.yaml --mode cloud
+```
+
+### Create the Nodes
+
+#### Create the Control Plane Nodes
+
+First a control plane needs to be created, with the example below creating 3 instances in a loop.
+The instance type (noted by the `--plan vc2-2c-4gb` argument) in the example is for a minimum-spec control plane node, and should be updated to suit the cluster being created.
+
+```bash
+for id in $(seq 3); do
+    vultr-cli instance create \
+        --plan vc2-2c-4gb \
+        --region $REGION \
+        --iso $TALOS_ISO_ID \
+        --host talos-k8s-cp${id} \
+        --label "Talos Kubernetes Control Plane" \
+        --tags talos,kubernetes,control-plane
```

Comment:
is it possible to pass the machine config as user data?

Reply:
Vultr supports user-data, but not when booting from a custom ISO it seems.